pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -308,6 +309,155 @@ func (am *Alertmanager) SyncAndApplyConfigFromDatabase() error {
 	return nil
 }
 
+func (am *Alertmanager) TestReceivers(ctx context.Context, c apimodels.TestReceiversConfig) (*apimodels.TestReceiversResult, error) {
+	// now represents the start time of the test
+	now := time.Now()
+	testAlert := &types.Alert{
+		Alert: model.Alert{
+			Labels: model.LabelSet{
+				model.LabelName("instance"): "foo",
+			},
+			Annotations: model.LabelSet{},
+			StartsAt:    now,
+		},
+		UpdatedAt: now,
+	}
+
+	// we must set a group key as this is required by some notification channels (i.e. webhook)
+	ctx = notify.WithGroupKey(ctx, testAlert.Labels.String())
+
+	tmpl, err := am.getTemplate()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get template: %w", err)
+	}
+
+	integrationsMap, err := am.buildIntegrationsMap(c.Receivers, tmpl)
+	if err != nil {
+		// The request contains invalid config for one or more contact points
+		return nil, fmt.Errorf("failed to create receivers: %w", err)
+	}
+
+	// job contains all metadata required to test a receiver
+	type job struct {
+		Config       *apimodels.PostableGrafanaReceiver
+		ReceiverName string
+		Notifier     notify.Integration
+	}
+
+	// result contains all metadata required to test a receiver
+	// and an error that is non-nil if the test fails
+	type result struct {
+		Config       *apimodels.PostableGrafanaReceiver
+		ReceiverName string
+		Error        error
+	}
+
+	workers := sync.WaitGroup{}
+	numWorkers := 10
```

Comment:
If we have less than 10 jobs wouldn't this continue to spawn goroutines that do nothing? 

Shouldn't this be something like:
```
	numWorkers := loadRuleGroupsConcurrency
	if numWorkers < len(jobs) {
		concurrency = len(jobs)
	}
```